### PR TITLE
etcdmain: set the flag's value of gateway from environment variables

### PR DESCRIPTION
**Pull request by yuqitao**
Related #10602

Please read https://github.com/etcd-io/etcd/blob/master/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: pkg/flags/flag.go [hunk 1]
```diff
@@ -49,16 +49,16 @@ func SetFlagsFromEnv(prefix string, fs *flag.FlagSet) error {
 }
 
 // SetPflagsFromEnv is similar to SetFlagsFromEnv. However, the accepted flagset type is pflag.FlagSet
-// and it does not do any logging.
-func SetPflagsFromEnv(prefix string, fs *pflag.FlagSet) error {
+// and the caller can choose to do logging or not.
+func SetPflagsFromEnv(prefix string, fs *pflag.FlagSet, log bool) error {
```
Comment (yuqitao) [inline pkg/flags/flag.go]: I change this, bacause package etcdctl doesn't do logging.

**Comment by yuqitao** [comment]
@hexfusion Should we do this for `grpc-proxy`?

**Comment by yuqitao** [comment]
@jingyih PTAL

**Comment by jingyih** [comment]
@yuqitao Could you help update the usage accordingly
https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/gateway.md

**Comment by yuqitao** [comment]
> @yuqitao Could you help update the documentation accordingly:
> https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/gateway.md

@jingyih PTAL

**Comment by jingyih** [comment]
Thanks for adding this feature.

@hexfusion could you also help take a look. This fixes #10602.

**Comment by MPV** [comment]
Anyone open to reviewing this?

**Review by spzala — COMMENTED** [review]
@yuqitao sorry this PR was overlooked earlier. The changes seems good overall. However, few things are now changed with repo. The master branch is renamed to main and so I couldn't reopen this PR. Also, the docs (e.g. gateway.md changes) are now moved to https://github.com/etcd-io/website So I think the best way going forward is to create a new PR. Thanks!

**Comment by spzala** [comment]
cc @MPV - related to your comment in the https://github.com/etcd-io/etcd/issues/10602

**Comment by yuqitao** [comment]
I haven't used etcd for a long time.  and I have no build environment.

**Pull request closed without merging**